### ISSUE TITLE
Websocket Timer Expired

### ISSUE THREAD

**Issue by iPerky** [steemit/steem#35]
getting this error on 3 of my 4 servers (x86_64)
```
steem-cli_wallet -s ws://localhost:8040 -H 127.0.0.1:8042 --rpc-http-allowip=127.0.0.1
1349736ms th_a       main.cpp:101                  main                 ] allowed_ips: ["127.0.0.1"] 
Logging RPC to file: logs/rpc/rpc.log
Starting a new wallet
1349736ms th_a       main.cpp:154                  main                 ] wdata.ws_server: ws://localhost:8040 
0 exception: unspecified
Timer Expired
    {"message":"Timer Expired"}
    asio  websocket.cpp:439 operator()

    {"uri":"ws://localhost:8040"}
    th_a  websocket.cpp:624 connect
```
cmake
```
--   iostreams
--   coroutine
** websocketpp

=========== Used Build Configuration =============

-- ENABLE_CPP11        = ON
-- BUILD_EXAMPLES      = OFF
-- BUILD_TESTS         = OFF

-- WEBSOCKETPP_ROOT    = /home/jason/tmp/git/steem/libraries/fc/vendor/websocketpp
-- WEBSOCKETPP_BIN     = /home/jason/tmp/git/steem/libraries/fc/vendor/websocketpp/bin
-- WEBSOCKETPP_LIB     = /home/jason/tmp/git/steem/libraries/fc/vendor/websocketpp/lib
-- Install prefix      = /usr/local

-- WEBSOCKETPP_BOOST_LIBS        = 
-- WEBSOCKETPP_PLATFORM_LIBS     = 
-- WEBSOCKETPP_PLATFORM_TSL_LIBS = 

-- Finished fc module configuration...
-- 

             CONFIGURED FOR STEEM NETWORK             


-- 

             CONFIGURED FOR LOW MEMORY NODE          


-- Configuring done
-- Generating done
```

Built using boost 1.57 - 1.58 and 1.60 on 3 different x86_64 servers ubuntu 15.10 and Archlinux.
How do I connect with the cli-wallet?

**Comment by mvandeberg** [comment]
What is your config file for steemd? Remove private keys please.

**Comment by iPerky** [comment]
config.ini
```
# Endpoint for P2P node to listen on
# p2p-endpoint = 

# Maxmimum number of incoming connections on P2P endpoint
# p2p-max-connections = 

# P2P nodes to connect to on startup (may specify multiple times)
# seed-node = 
seed-node = 52.38.66.234:2001
seed-node = 52.37.169.52:2001
seed-node = 52.26.78.244:2001
seed-node = 54.194.17.248:2001
seed-node = 104.168.154.160:40696   
seed-node = 212.47.249.84:40696     

# Pairs of [BLOCK_NUM,BLOCK_ID] that should be enforced as checkpoints.
# checkpoint = 

# Endpoint for websocket RPC to listen on
# rpc-endpoint = 
rpc-endpoint = 127.0.0.1:8040

# Endpoint for TLS websocket RPC to listen on
# rpc-tls-endpoint = 

# The TLS certificate file for this server
# server-pem = 

# Password for this certificate
# server-pem-password = 

# Block signing key to use for init witnesses, overrides genesis file
# dbg-init-key = 

# API user specification, may be specified multiple times
# api-user = 

# Set an API to be publicly available, may be specified multiple times
public-api = database_api login_api

# Plugin(s) to enable, may be specified multiple times
enable-plugin = witness account_history

# Defines a range of accounts to track as a json pair ["from","to"] [from,to)
# track-account-range = 

# RPC endpoint of a trusted validating node (required)
# trusted-node = 

# Enable block production, even if the chain is stale.
enable-stale-production = false

# Percent of witnesses (0-99) that must be participating in order to produce blocks
required-participation = false

# name of witness controlled by this node (e.g. initwitness )
# witness = 

# name of miner and its private key (e.g. ["account","WIF PRIVATE KEY"] )
# miner = 

# Number of threads to use for proof of work mining
# mining-threads = 

# WIF PRIVATE KEY to be used by one or more witnesses or miners
# private-key = 

# Account creation fee to be voted on upon successful POW - Minimum fee is 100.000 STEEM (written as 100000)
# miner-account-creation-fee = 

# Maximum block size (in bytes) to be voted on upon successful POW - Max block size must be between 128 KB and 750 MB
# miner-maximum-block-size = 

# SBD interest rate to be vote on upon successful POW - Default interest rate is 10% (written as 1000)
# miner-sbd-interest-rate = 

# declare an appender named "stderr" that writes messages to the console
[log.console_appender.stderr]
stream=std_error

# declare an appender named "p2p" that writes messages to p2p.log
[log.file_appender.p2p]
filename=logs/p2p/p2p.log
# filename can be absolute or relative to this config file

# route any messages logged to the default logger to the "stderr" logger we
# declared above, if they are info level are higher
[logger.default]
level=warn
appenders=stderr

# route messages sent to the "p2p" logger to the p2p appender declared above
[logger.p2p]
level=warn
appenders=p2p


```

**Comment by iPerky** [comment]
This issue could be self-inflicted as I am getting the same error using muse and bitshares.
I have disabled ufw and rebuilt using DEBUG.
A package update may have broken websockets.

Its very strange that it has happened on different servers using different OSs and boosts.
steemd start up looks like it should work:
```
steemd
------------------------------------------------------

            STARTING STEEM NETWORK

------------------------------------------------------
initminer public key: STM8GC13uCZbP44HzMLV6zPZGwVQ8Nt4Kji8PapsPiNq1BK153XTX
chain id: 0000000000000000000000000000000000000000000000000000000000000000
------------------------------------------------------
2341530ms th_a       main.cpp:118                  main                 ] Error parsing logging config from config file /home/jason/steem/witness_node_data_dir/config.ini, using default config
2341530ms th_a       main.cpp:157                  main                 ] parsing options
2341530ms th_a       main.cpp:159                  main                 ] initializing node
2341530ms th_a       main.cpp:161                  main                 ] initializing plugins
2341530ms th_a       application.cpp:977           initialize_plugins   ] Initializing plugin account_history
2341530ms th_a       application.cpp:977           initialize_plugins   ] Initializing plugin witness
2341530ms th_a       witness.cpp:95                plugin_initialize    ] witness plugin:  plugin_initialize() begin
2341530ms th_a       witness.cpp:98                plugin_initialize    ] _witnesses: ["ihashfury"] 
2341562ms th_a       witness.cpp:164               plugin_initialize    ] witness plugin:  plugin_initialize() end
2341563ms th_a       main.cpp:164                  main                 ] starting node
2342837ms th_a       database.cpp:87               open                 ] last_block->id(): 000ee2e087c501d7a6e4388a849a27c95586b89a last_block->block_num(): 975584 
2342837ms th_a       thread.cpp:95                 thread               ] name:ntp tid:140525478262528
2342838ms th_a       application.cpp:345           startup              ] API database_api enabled publicly
2342838ms th_a       application.cpp:345           startup              ] API login_api enabled publicly
2342838ms th_a       thread.cpp:95                 thread               ] name:p2p tid:140525461477120
2342839ms th_a       application.cpp:99            reset_p2p_node       ] Adding seed node 52.38.66.234:2001
2342839ms th_a       application.cpp:99            reset_p2p_node       ] Adding seed node 52.37.169.52:2001
2342839ms th_a       application.cpp:99            reset_p2p_node       ] Adding seed node 52.26.78.244:2001
2342839ms th_a       application.cpp:99            reset_p2p_node       ] Adding seed node 54.194.17.248:2001
2342839ms th_a       application.cpp:99            reset_p2p_node       ] Adding seed node 104.168.154.160:40696
2342840ms th_a       application.cpp:99            reset_p2p_node       ] Adding seed node 212.47.249.84:40696
2342840ms th_a       application.cpp:121           reset_p2p_node       ] Configured p2p node to listen on 0.0.0.0:41684
2342840ms th_a       application.cpp:177           reset_websocket_serv ] Configured websocket rpc to listen on 127.0.0.1:8040
2342841ms th_a       main.cpp:166                  main                 ] starting plugins
2342841ms th_a       witness.cpp:169               plugin_startup       ] witness plugin:  plugin_startup() begin
2342841ms th_a       witness.cpp:176               plugin_startup       ] Launching block production for 1 witnesses.
2342841ms th_a       witness.cpp:194               plugin_startup       ] No miners configured! Please add miner names and private keys to configuration.
2342841ms th_a       witness.cpp:196               plugin_startup       ] witness plugin:  plugin_startup() end
2342841ms th_a       main.cpp:181                  main                 ] Started witness node on a chain with 975584 blocks.
2382112ms th_a       application.cpp:417           handle_block         ] Syncing Blockchain --- Got block: #980000 time: 2016-04-28T11:23:18
```

**Comment by iPerky** [comment]
same issue on v0.4.0

Packages
#ubuntu
```
sudo apt-get install gcc g++ cmake make libbz2-dev libdb++-dev libdb-dev libssl-dev openssl libreadline-dev autoconf libtool git libncurses-dev build-essential kernel-package libncurses-dev doxygen libboost-all-dev autotools-dev automake libminiupnpc-dev libwebsocketpp-dev libboost-dev libicu-dev python-dev uuid-dev ncurses-dev
```

#archlinux
```
pacman -S git base-devel cmake gcc boost make automake doxygen ncurses readline
```

#build script
```
git pull
#git checkout develop
# v0.4.0
git checkout master
git describe --tags 
make clean
git submodule update --init --recursive	
cmake -DENABLE_CONTENT_PATCHING=OFF -DLOW_MEMORY_NODE=ON -D CMAKE_BUILD_TYPE=Release .
nice -n 19 make steemd cli_wallet -j 2
git describe --tags
date
```
cmake and make build without issues

#strip backup & copy
```
strip programs/steemd/steemd
strip programs/cli_wallet/cli_wallet

cp ~/bin/steemd ~/bin/steemd.bak
cp  programs/steemd/steemd ~/bin

cp ~/bin/steem-cli_wallet ~/bin/steem-cli_wallet.bak
cp programs/cli_wallet/cli_wallet ~/bin/steem-cli_wallet
git describe --tags
```
cli error on Ubuntu 15.01 and Archlinux
```
steem-cli_wallet -s ws://127.0.0.1:8040
Logging RPC to file: logs/rpc/rpc.log
Starting a new wallet
2054707ms th_a       main.cpp:154                  main                 ] wdata.ws_server: ws://127.0.0.1:8040 
0 exception: unspecified
Timer Expired
    {"message":"Timer Expired"}
    asio  websocket.cpp:439 operator()

    {"uri":"ws://127.0.0.1:8040"}
    th_a  websocket.cpp:624 connect

```
I have also used server port 8090 and disabled ufw but get the same error.
Any help is much appreciated.

**Comment by iPerky** [comment]
Update to ufw changed my iptable rules.
Purged ufw and resolved the issue.

**Comment by Percise** [comment]
what is ufw? explain process

**Comment by Netherdrake** [comment]
I have the same issue with new builds (used to work OK). Disabling ufw and reseting networkmanager didn't fix the issue.

**Comment by roelandp** [comment]
fix: update this file: 
https://github.com/zaphoyd/websocketpp/commit/0bb33e4bca4ccc42a36aa2321e4fb97f2562e519

Will be fixed in websocket++ 0.8 which hopefully gets included in Steem. 

Github Discussion to learn more: 
https://github.com/zaphoyd/websocketpp/issues/623

I did a manual change to the file mentioned above and then it works. You will need to `cmake / make` again. 

Apparently this is a 'change' in Linux Kernel > 4.4 so first check with `uname -r` that you are indeed running a node with kernel > 4.4
